system/logaccess/ElasticStack/ElasticStackLogAccess.cpp

```diff
@@ -520,11 +611,69 @@ cpr::Response ElasticStackLogAccess::performESQuery(const LogAccessConditions &
 bool ElasticStackLogAccess::fetchLog(const LogAccessConditions & options, StringBuffer & returnbuf, LogAccessLogFormat format)
 {
     cpr::Response esresp = performESQuery(options);
-    processESJsonResp(esresp, returnbuf, format);
+    processESSearchJsonResp(esresp, returnbuf, format);
 
     return true;
 }
 
+class ELASTICSTACKLOGACCESS_API ElasticStackLogStream : public CInterfaceOf<IRemoteLogAccessStream>
+{
+public:
+    virtual bool readLogEntries(StringBuffer & record, unsigned & recsRead) override
+    {
+        Json::Value res;
+        recsRead = 0;
+
+        if (m_esSroller.next(res))
+        {
+            if (!res["hits"].empty())
+            {
+                recsRead = res["hits"].size();
+                std::ostringstream sout;
+                m_jsonWriter->write(res, &sout); // serialize Json object to string for processing
+                processESScrollJsonResp(sout.str().c_str(), record, m_outputFormat, false); // convert Json string to target format
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    ElasticStackLogStream(std::string & queryString, const char * connstr, const char * indexsearchpattern, LogAccessLogFormat format,  std::size_t pageSize, std::string scrollTo)
+     : m_esSroller(std::make_shared<elasticlient::Client>(std::vector<std::string>({connstr})), pageSize, scrollTo)
+    {
+        m_outputFormat = format;
+        m_esSroller.init(indexsearchpattern, DEFAULT_ES_DOC_TYPE, queryString);
+        m_jsonWriter.reset(m_jsonStreamBuilder.newStreamWriter());
+    }
+
+    virtual ~ElasticStackLogStream() override
+    {
+        m_esSroller.clear();
```

Comment:
trivial: Do these need to be explict?

Reply:
It seemed like the correct approach, can you expand your thoughts on this?

Reply:
Because the default destructor would do this anyway?